examples/step-78/step-78.cc

```diff
@@ -0,0 +1,869 @@
+#include <deal.II/base/convergence_table.h>
+
+#include <fstream>
+#include <iostream>
+
+namespace BlackScholesSolver
+{
+  using namespace dealii;
+
+  template <int dim>
+  double coefficient(const Point<dim> &p)
+  {
+    return p.square();
+  }
+
+  template <int dim>
+  Tensor<1, dim> SCoefficient(const Point<dim> &p)
+  {
+    return Tensor<1, dim>(p);
+  }
+
+  /*
+  This Section creates a class for the known solution when testing using the MMS
+  I am using $v(\tau,S) = -\tau^2 -S^2 + 6$ for my solution
+  */
+  template <int dim>
+  class Solution : public Function<dim>
+  {
+  public:
+    Solution(double __maturity_time);
+    virtual double value(const Point<dim> & p,
+                         const unsigned int component = 0) const override;
+
+    virtual Tensor<1, dim>
+    gradient(const Point<dim> & p,
+             const unsigned int component = 0) const override;
+
+    Tensor<1, dim> time_deriv(const Point<dim> & p,
+                              const unsigned int component = 0) const;
+
+  private:
+    double _maturity_time;
+  };
+
+
+  template <int dim>
+  double Solution<dim>::value(const Point<dim> & p,
+                              const unsigned int component) const
+  {
+    double return_value = 0;
+    return_value +=
+      -std::pow(p(component), 2) - std::pow(this->get_time(), 2) + 6;
+    return return_value;
+  }
+
+  template <int dim>
+  Tensor<1, dim> Solution<dim>::gradient(const Point<dim> & p,
+                                         const unsigned int component) const
+  {
+    Tensor<1, dim> return_value;
+    return_value = Point<1>(-2 * p(component));
+    return return_value;
+  }
+
+  template <int dim>
+  Tensor<1, dim> Solution<dim>::time_deriv(const Point<dim> & p,
+                                           const unsigned int component) const
+  {
+    Tensor<1, dim> return_value;
+    for (int i = 0; dim; i++)
+      {
+        return_value += -2 * (this->get_time()) * std::sin(p(i)) * Point<1>(1);
+      }
+    return return_value;
+  }
+
+  /*
+  The next piece is the declaration of the main class of this program. This is
+  very similar to the Step-26 tutorial, with some modifications. New matrices
+  had to be added to calculate the A and B matrices, as well as the $V_{diff}$
+  vector mentioned in the introduction.
+  */
+  template <int dim>
+  class BlackScholes
+  {
+  public:
+    BlackScholes();
+
+    // enum RefinementMode
+    // {
+    //     global_refinement,
+    //     adaptive_refinement
+    // };
+
+    void run();
+
+    const double s_max;
+    const double maturity_time;
+    const double sigma;
+    const double r;
+    const double strike_price;
+
+    // RefinementMode refinement_mode;
+
+  private:
+    void setup_system();
+    void solve_time_step();
+    void solve_time_step_diffusion();
+    void refine_mesh(const unsigned int min_grid_level,
+                     const unsigned int max_grid_level);
+    void refine_grid();
+    void o_results();
```

Comment:
```suggestion
    void o_results();
```

Reply:
I'm not sure what the change is here.

Reply:
I don't know either, but maybe name the function `output_results()` like in all other programs.